src/features/room/helpers/join.tsx

```diff
@@ -34,42 +32,26 @@ const Abi = {
 }
 
 interface Options {
-    onToast?: (props: ComponentProps<typeof Toast>) => void
+    retoastConstroller?: RetoastController
 }
 
 export default function join(
     stream: Stream,
     requester: Address,
-    { onToast: onToastProp }: Options = {}
+    { retoastConstroller }: Options = {}
 ) {
     return call(function* () {
         const roomId = stream.id
 
         const { name, tokenAddress, tokenType, stakingEnabled = false } = getRoomMetadata(stream)
 
-        let tc: Controller | undefined
+        const toast = retoastConstroller || retoast()
```

Comment:
This is interesting. We reuse the outer toast, or use a new, internal one. Nifty UX, better than before.